tests/AwaitTest.php

```diff
@@ -10,6 +10,30 @@
 
 class AwaitTest extends TestCase
 {
+    /**
+     * @dataProvider provideAwaiters
+     */
+    public function testExecutionOrder(callable $await)
+    {
+        self::expectOutputString('acbd');
+
+        $deferred = new React\Promise\Deferred();
+
+        $promises[] = React\Async\async(function () use ($deferred, $await) {
+            print 'a';
+            $await($deferred->promise());
+            print 'b';
+        })();
+
+        $promises[] = React\Async\async(function () use ($deferred) {
+            print 'c';
+            $deferred->resolve();
+            print 'd';
+        })();
```

Comment:
In my opinion it makes more sense to expect an `abcd` outcome instead of adjusting the outcome to the written test.
```suggestion
        self::expectOutputString('abcd');

        $deferred = new React\Promise\Deferred();

        $promises[] = React\Async\async(function () use ($deferred, $await) {
            print 'a';
            $await($deferred->promise());
            print 'c';
        })();

        $promises[] = React\Async\async(function () use ($deferred) {
            print 'b';
            $deferred->resolve();
            print 'd';
        })();
```